richtext-ui/src/commonMain/kotlin/com/halilibo/richtext/ui/string/Text.kt

```diff
@@ -68,6 +154,77 @@ public fun RichTextScope.Text(
   }
 }
 
+private data class TextAnimation(val startIndex: Int, val alpha: Float)
+
+private fun AnnotatedString.animateAlphas(animations: Collection<TextAnimation>, contentColor: Color): AnnotatedString {
+  if (this.text.isEmpty() || animations.isEmpty()) {
+    return this
+  }
+  var remainingText = this
+  val modifiedTextSnippets = mutableStateListOf<AnnotatedString>()
+  animations.sortedByDescending { it.startIndex }.forEach { animation ->
+    if (animation.startIndex < remainingText.length) {
+      modifiedTextSnippets.add(
+        remainingText.subSequence(animation.startIndex, remainingText.length)
+          .changeAlpha(animation.alpha, contentColor)
+      )
+      remainingText = remainingText.subSequence(0, animation.startIndex)
+    } else {
+    }
+  }
+  return AnnotatedString.Builder(remainingText).apply {
+    modifiedTextSnippets.reversed().forEach { append(it) }
+  }.toAnnotatedString()
+//  return modifiedTextSnippets.reversed().fold(remainingText) { acc, snippet -> acc + snippet }
```

Comment:
Did you mean to leave this commented?

Reply:
yeah sorry, one is an optimization of the other